src/languageclient.rs

```diff
@@ -117,6 +117,10 @@ impl State {
                 "has('nvim')",
             ].as_ref(),
         )?;
+
+        let (diagnosticsMaxSigns,): (i64,) =
```

Comment:
Please merge this with the previous lines. Every `eval` is an RPC call and thus has overhead.

Reply:
The size of tuple that `serde_json::from_value<T>` can interpret was up to 16, and I could not merge the variable with previous `let` lines (because already 16 variables exist in the `let` lines).
I moved the variables into a struct and assigned `serde_json::from_value<T>` as the struct. Now, the overhead was fixed.

Ref:
1. `serde_json::from_value<T>` calls `T::deserialize`: https://docs.serde.rs/src/serde_json/value/mod.rs.html#1121-1126
2. The defined tuple size of `T` is up to 16: https://docs.serde.rs/serde/trait.Deserialize.html

Reply:
Ahh, sorry, I didn't know about this limitation.

Please use the original separate RPC call. (The current implementation obviously misses some properties when syncing to global `state`.)

Reply:
Thank you for the review. I updated the commit.

> The current implementation obviously misses some properties when syncing to global state.

I could not find what properties was missed. Could you pointed out some of the properties?
(This is my first rust program, and I'm sorry if I made an easy mistake.)
The previous commit is [this](https://github.com/blyoa/LanguageClient-neovim/commit/4ad3f3850b648992c5bdac53ebe10a6df0784092).

Reply:
My bad. I missed that some of the settings are copied from the intermediate `vimSettings` to local variables.

I noticed you updated the PR. Should be good to go.